apps/api/src/app/events/usecases/map-trigger-recipients/map-trigger-recipients.e2e.ts

```diff
@@ -0,0 +1,275 @@
+import { expect } from 'chai';
+import { v4 as uuid } from 'uuid';
+
+import { MapTriggerRecipients } from './map-trigger-recipients.use-case';
+import { MapTriggerRecipientsCommand } from './map-trigger-recipients.command';
+
+import { SharedModule } from '../../../shared/shared.module';
+import { EventsModule } from '../../events.module';
+
+const createTopicEntity = async (
+  session: UserSession,
+  topicRepository: TopicRepository,
+  topicSubscribersRepository: TopicSubscribersRepository,
+  topicKey: TopicKey,
+  topicName: TopicName
+): Promise<TopicEntity> => {
+  const environmentId = session.environment._id;
+  const organizationId = session.organization._id;
+  const userId = session.user._id;
+
+  const topicEntity = {
+    _environmentId: TopicRepository.convertStringToObjectId(environmentId),
+    key: topicKey,
+    name: topicName,
+    _organizationId: TopicRepository.convertStringToObjectId(organizationId),
+    _userId: TopicRepository.convertStringToObjectId(userId),
+  };
+  const topic = await topicRepository.create(topicEntity);
+
+  const topicSubscribersEntity = {
+    _environmentId: TopicRepository.convertStringToObjectId(environmentId),
+    _topicId: topic._id,
+    _organizationId: TopicRepository.convertStringToObjectId(organizationId),
+    _userId: TopicRepository.convertStringToObjectId(userId),
+  };
+
+  const topicSubscribers = await topicSubscribersRepository.create(topicSubscribersEntity);
+
+  expect(topic).to.exist;
+  expect(topic.key).to.be.eql(topicKey);
+  expect(topic.name).to.be.eql(topicName);
+
+  return topic;
+};
+
+const addSubscribersToTopic = async (
+  session: UserSession,
+  topicRepository: TopicRepository,
+  topicSubscribersRepository: TopicSubscribersRepository,
+  topicId: TopicId,
+  subscribers: SubscriberId[]
+): Promise<void> => {
+  const _environmentId = TopicSubscribersRepository.convertStringToObjectId(session.environment._id);
+  const _organizationId = TopicSubscribersRepository.convertStringToObjectId(session.organization._id);
+  const _topicId = TopicSubscribersRepository.convertStringToObjectId(topicId);
+  const _userId = TopicSubscribersRepository.convertStringToObjectId(session.user._id);
+
+  const entity: TopicSubscribersEntity = {
+    _environmentId,
+    _organizationId,
+    _topicId,
+    _userId,
+    subscribers,
+  };
+  await topicSubscribersRepository.addSubscribers(entity);
+
+  const result = await topicRepository.findTopic({
+    _environmentId,
+    _organizationId,
+    _id: _topicId,
+    _userId,
+  });
+
+  expect(result.subscribers).to.be.eql(subscribers);
+};
+
+const buildCommand = (
+  session: UserSession,
+  transactionId: string,
+  recipients: TriggerRecipientsPayload
+): MapTriggerRecipientsCommand => {
+  return MapTriggerRecipientsCommand.create({
+    organizationId: session.organization._id,
+    environmentId: session.environment._id,
+    recipients,
+    transactionId,
+    userId: session.user._id,
+  });
+};
```

Comment:
I find myself in the opposite position. I find more readable from top to bottom having the helper functions on top of the file. I think this comes from my scripting languages and non strict mode issues experience that developed into this habit. 
@ainouzgali already asked me in a previous PR for the same change. So if it will be a standard decision on our codebase I am ok with changing it.

Reply:
I do also prefer to have helpers at the top ;) To me, the natural way to define them is at the top because that's what the JS engine does underneath.